src/Http/Routing/src/ShortCircuit/RouteShortCircuitEndpointRouteBuilderExtensions.cs

```diff
@@ -0,0 +1,62 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using Microsoft.AspNetCore.Builder;
+using Microsoft.AspNetCore.Http;
+
+namespace Microsoft.AspNetCore.Routing;
+
+/// <summary>
+/// Provides extension methods for <see cref="IEndpointRouteBuilder"/> to add short circuited endpoints.
+/// </summary>
+public static class RouteShortCircuitEndpointRouteBuilderExtensions
+{
+    private static readonly RequestDelegate _shortCircuitDelegate = (context) => Task.CompletedTask;
+    /// <summary>
+    /// Adds a <see cref="RouteEndpoint"/> to the <see cref="IEndpointRouteBuilder"/> that matches HTTP GET requests
+    /// for the specified prefixes.
+    /// </summary>
+    ///<param name="builder">The <see cref="IEndpointRouteBuilder"/> to add the route to.</param>
+    /// <param name="statusCode">The status code to set in the response.</param>
+    /// <param name="routePrefixes">An array of route prefixes to be short circuited.</param>
+    /// <returns>A <see cref="IEndpointConventionBuilder"/> that can be used to further customize the endpoint.</returns>
+    public static IEndpointConventionBuilder MapShortCircuit(this IEndpointRouteBuilder builder, int statusCode, params string[] routePrefixes)
+    {
+        var group = builder.MapGroup("");
+        foreach (var routePrefix in routePrefixes)
+        {
+            string route;
+            if (routePrefix.EndsWith("/", StringComparison.OrdinalIgnoreCase))
+            {
+                route = $"{routePrefix}{{**catchall}}";
+            }
+            else
+            {
+                route = $"{routePrefix}/{{**catchall}}";
+            }
+            group.Map(routePrefix, _shortCircuitDelegate).ShortCircuit(statusCode);
+        }
+
+        return new EndpointConventionBuilder(group);
```

Comment:
Why not return the group?

Reply:
> I agree groups are convenient for this. The one problem I see is that even though the declared type is IEndpointConventionBuilder, someone could call (RouteGroupBuilder)app.MapShortCircuit(404, "/prefix1", "/prfix2") and then add additional routes.

> While not exactly a problem, it's a little weird, and would prevent us from returning another IEndpointConventionBuilder that's not derived from the currently-sealed RouteGroupBuilder without going through the breaking change process since it'd be considered a [behavioral breaking change](https://learn.microsoft.com/en-us/dotnet/core/compatibility/library-change-rules#behavioral-changes) to the return value. So probably better to be safe and copy the private [CompositeEndpointConventionBuilder](https://github.com/dotnet/aspnetcore/blob/33e7a9a111742c8b06924383baba5c448d7f8af8/src/SignalR/common/Http.Connections/src/ConnectionEndpointRouteBuilderExtensions.cs#L131). Maybe we could make it shared source.

From https://github.com/dotnet/aspnetcore/pull/46713#discussion_r1115138619

Reply:
@halter73 is this really a concern?

Reply:
Sorry I missed this, but yes. If we think it's valuable to return a RouteGroupBuilder, we should make that the declared return type. You know people will end up casting it anyway if we let them.

Reply:
The current implementation is fine, if cautious. We can decide to remove the wrapper later if we think that caution is not needed. I don't think we want to change the API return type though. Publicly returning the group exposes an implementation detail and sets the wrong expectation about how that return type can be used. The group APIs don't make sense for this scenario. We only return the builder so that other metadata can be added like http method constraints.